File: DesignPattenStudy/CommandPattern/Java/Rider.java
package CommandPattern.Java;

public class Rider {

    private DeliveryCommand deliveryCommand;

    public Rider(DeliveryCommand deliveryCommand) {

        this.deliveryCommand = deliveryCommand;
    }

    public Rider changeDeliveryCommand(DeliveryCommand deliveryCommand) {

        this.deliveryCommand = deliveryCommand;
        return this;
    }

    public String delivery() {

        return deliveryCommand.deliveryVehicle();
    }
}

File: DesignPattenStudy/CommandPattern/Kotlin/Rider.kt
package CommandPattern.Kotlin

class Rider(private var deliveryCommand: DeliveryCommand) {

    fun changeDeliveryCommand(deliveryCommand: DeliveryCommand): Rider {
        this.deliveryCommand = deliveryCommand
        return this
    }

    fun delivery(): String = this.deliveryCommand.driveVehicle()
}